### ISSUE TITLE
test: box/sql.test.lua sometimes fails

### ISSUE THREAD

**Issue by ImeevMA** [tarantool/tarantool#3737]
Error:
```bash
[001] Test failed! Result content mismatch:
[001] --- box/sql.result	Wed Oct 10 18:51:24 2018
[001] +++ box/sql.reject	Thu Oct 11 17:36:11 2018
[001] @@ -12,12 +12,14 @@
[001]  ...
[001]  _ = box.schema.space.create('test1', { id = 555 })
[001]  ---
[001] +- error: Duplicate key exists in unique index 'primary' in space '_space'
[001]  ...
[001]  box.schema.user.create('test', { password = 'test' })
[001]  ---
[001]  ...
[001]  box.schema.user.grant('test', 'read,write,alter', 'space', 'test1')
[001]  ---
[001] +- error: Space 'test1' does not exist
[001]  ...
[001]  box.schema.user.grant('test', 'read,write,alter', 'space', 'test')
[001]  ---
[001] @@ -33,9 +35,13 @@
[001]  ...
[001]  _ = box.space.test1:create_index('primary', { type = 'hash' })
[001]  ---
[001] +- error: '[string "_ = box.space.test1:create_index(''primary'', {..."]:1: attempt
[001] +    to index field ''test1'' (a nil value)'
[001]  ...
[001]  _ = box.space.test1:create_index('secondary', { type = 'hash', parts = {2, 'string'}})
[001]  ---
[001] +- error: '[string "_ = box.space.test1:create_index(''secondary'',..."]:1: attempt
[001] +    to index field ''test1'' (a nil value)'
[001]  ...
[001]  --  send request to remote server to force schema reloading
[001]  conn:reload_schema()
[001] @@ -194,15 +200,18 @@
[001]  -- check update delete be secondary index
[001]  conn.space.test1:insert{0, "hello", 1}
[001]  ---
[001] -- [0, 'hello', 1]
[001] +- error: '[string "return conn.space.test1:insert{0, "hello", 1} "]:1: attempt to
[001] +    index field ''test1'' (a nil value)'
[001]  ...
[001]  conn.space.test1.index.secondary:update("hello", {{'=', 3, 2}})
[001]  ---
[001] -- [0, 'hello', 2]
[001] +- error: '[string "return conn.space.test1.index.secondary:updat..."]:1: attempt to
[001] +    index field ''test1'' (a nil value)'
[001]  ...
[001]  conn.space.test1.index.secondary:delete("hello")
[001]  ---
[001] -- [0, 'hello', 2]
[001] +- error: '[string "return conn.space.test1.index.secondary:delet..."]:1: attempt to
[001] +    index field ''test1'' (a nil value)'
[001]  ...
[001]  -- cleanup 
[001]  space:delete(0)
[001] @@ -217,6 +226,8 @@
[001]  ...
[001]  box.space.test1:drop()
[001]  ---
[001] +- error: '[string "return box.space.test1:drop() "]:1: attempt to index field ''test1''
[001] +    (a nil value)'
[001]  ...
[001]  box.schema.user.drop('test')
[001]  ---
[001] 
```

Reproduce .yaml:
```yaml
- [box/stat.test.lua, null]
- [box/rtree_misc.test.lua, null]
- [box/func_reload.test.lua, null]
- [box/reconfigure.test.lua, null]
- [box/transaction.test.lua, null]
- [box/blackhole.test.lua, null]
- [box/rtree_point_r2.test.lua, null]
- [box/iterator.test.lua, null]
- [box/tree_pk.test.lua, null]
- [box/sequence.test.lua, null]
- [box/sql.test.lua, null]
```

**Comment by Totktonada** [comment]
> _ = box.schema.space.create('test1', { id = 555 })

2.0 has no {id = 555} here. Are you able to reproduce it on a clean 2.0?

**Comment by locker** [comment]
Yep, turns out I've already fixed this issue, see commit 23e71c6e4156294db74e830290e59f10616c509c.
